packages/block-library/src/navigation/index.php

```diff
@@ -1620,7 +1620,12 @@ function block_core_navigation_update_ignore_hooked_blocks_meta( $post ) {
  * Injection of hooked blocks into the Navigation block relies on some functions present in WP >= 6.5
  * that are not present in Gutenberg's WP 6.5 compatibility layer.
  */
-if ( function_exists( 'set_ignored_hooked_blocks_metadata' ) && ! has_filter( 'rest_pre_insert_wp_navigation', $rest_insert_wp_navigation_core_callback ) ) {
+if (
+	function_exists( 'set_ignored_hooked_blocks_metadata' ) && (
+		! has_filter( 'rest_pre_insert_wp_navigation', $rest_insert_wp_navigation_core_callback ) ||
+		! has_filter( 'rest_pre_insert_wp_navigation', 'update_ignored_hooked_blocks_postmeta' )
+	)
+) {
```

Comment:
Like so?

```suggestion
if (
	! has_filter( 'rest_pre_insert_wp_navigation', 'update_ignored_hooked_blocks_postmeta' ) &&
	function_exists( 'set_ignored_hooked_blocks_metadata' ) &&
	! has_filter( 'rest_pre_insert_wp_navigation', $rest_insert_wp_navigation_core_callback )
) {
```

So we add the filter here if

- Core hasn't added `update_ignored_hooked_blocks_postmeta`, and
- there's a `set_ignored_hooked_blocks_metadata` function (meaning that we're running on WP 6.5), and
- Core hasn't added `$rest_insert_wp_navigation_core_callback`.

Seem about right? 🤔

Reply:
Ah you've saved me a commit there. I was about to commit this myself. Yes you're right, I came up with the same condition myself locally.

Reply:
Great minds! 🧠🧠 😆